src/pages/PlansList.tsx

```diff
@@ -43,62 +47,109 @@ const PlansList = () => {
   })
   let index = -1
 
+  useEffect(() => {
```

Comment:
So considering how it's used and that you repeat the logic in all the places with search, I think we need to architecture it differently : 

`TextInputForSearch` doesn't really make sense, maybe just as a UI component and then it should accept the prop "onChange"
Also the `TextInput` already has a `localValue` so I don't think you need an additional one here 🤔

Then all the logic regarding the query should be in `useDebouncedSearch` that should be used where the search is used : 
It should accept a `searchQuery` and `loading` and : 
- Launch the query on first load
- Return a `loading` state that is "debounced" has you're doing it here

Reply:
Changes have been pushed, thanks for your message